### ISSUE TITLE
[Question] Is there a way to use different authentication state for different workers

### ISSUE THREAD

**Issue by jithinjosejacob** [microsoft/playwright#14659]
I am running tests against a website using 3 workers

I am reusing the same authentication state to login

so when tests are run in parallel , same user gets logged in at 3 places which slows down the app.

Is there a chance that I can create 3 different authentication states using 3 different users and use each of them on 3 seperate workers so that they can login simultaneously

**Comment by pavelfeldman** [comment]
Yes, you can log in 3 times and pass 3 different storage states to 3 different projects that will run in parallel. See https://playwright.dev/docs/test-parameterize#parameterized-projects!

**Comment by jithinjosejacob** [comment]
can i do it for the same project ,lets say i want to use 3 workers for chrome. We are already using different storage state for different browsers

**Comment by pavelfeldman** [comment]
The structure of the tests should not depend on the number of workers - you should be able to run the same test suite with 1 worker and with 100. So I assume you want to sign in in each worker separately. You can achieve that via a worker-scope fixture. Something similar to (but not exactly like) this: https://playwright.dev/docs/next/test-fixtures#worker-scoped-fixtures

In the worker-scope fixture you would sign in, save the storageState into a file using testInfo.workerIndex (this is unique for each worker), then override storageState fixture to return respective file:

`// fixtures.ts`
```js
import { test as baseTest } from '@playwright/test';
export { expect } from '@playwright/test';

export const test = baseTest.extend({
  storageState: [async ({ browser }, use, testInfo) => {
    // Override storage state, use worker index to look up logged-in info and generate it lazily.
    const fileName = path.join(testInfo.project.outputDir, 'storage-' + testInfo.workerIndex);
    if (!fs.existsSync(fileName)) {
      const page = await browser.newPage()
      await page.goto('https://playwright.dev');
      await page.context().storageState({ path: fileName });
      await page.close();
    }
    await use(fileName);
  }],
});
```

then import your `test` via that file instead of from @playwright/test.

**Comment by pavelfeldman** [comment]
Assigning to 1.24 to add the above to the docs.

**Comment by jithinjosejacob** [comment]
Thanks . I will try this out
